wordpress-mobile/WordPress-iOS
# No feedback to users when creating Post + Camera is used to capture photos and Photo library permission not given

This PR adds functionality so that when a user tries to use their camera to capture a photo and add it to their post but has denied photo permission they get notified with an error message

Ref #17467

To test
	1. Clean install WordPress (delete first, then install)
	2. Confirm that Settings -> WordPress -> "Allow WordPress to Access" -> Doesn't contain anything about photos
	3. In a new Post, tap the + and choose the Image block
	4. Select "Take a Photo"
	5. When prompted with "'WordPress' Would Like to Access the Camera", tap "OK"
	6. Take a picture and choose "Use Photo"
	7. Tap "Don't allow" on the "'WordPress' Would Like to Access your Photos" prompt
	8. Error message is displayed to user informing them the app needs permission to access device media library

Note that in step 7 if the user chooses "Select Photo" the native OS iOS photo picker does not have the captured photo as an option.  This PR does present an error message to the user at the end of this flow but its not time appropriate.  The change required for this would be much more invasive and should be tackled in a different PR

Associated MediaPicker-iOS PR: https://github.com/wordpress-mobile/MediaPicker-iOS/pull/383

## Regression Notes
1. Potential unintended areas of impact
Permissions when capturing photo via camera

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
